### DISCUSSION TITLE
NavigationRail for MaterialDesign3 ListBox

### DISCUSSION THREAD

**Discussion by StefanoRivolta-Previero** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2682] [General]
I've updated the ListBox style to match the new MaterialDesign 3 guidelines for the NavigationRail. You can find the changes in this fork:

https://github.com/StefanoRivolta-Previero/MaterialDesignInXamlToolkit/tree/listbox  

## [Material Design Guidelines](https://m3.material.io/components/navigation-rail/specs):

<img width="676" alt="Light" src="https://user-images.githubusercontent.com/58171461/153855441-c0479ee3-1d48-4830-8e22-674c4ba3a49a.png"><img width="1137" alt="Dark" src="https://user-images.githubusercontent.com/58171461/153855444-3043c813-532e-4651-bc40-1148cf9fc772.png">

## Result:
![vcADXe0g21](https://user-images.githubusercontent.com/58171461/153855650-2e80bfaf-ae0d-4d50-9c82-5da913a11864.png)

The process to create an Item is a bit complex, since it needs two icons (one filled for when it's selected, and one outlined for when it's unselected) and a text under the selection border. 
I've opted for the content to only be the text part, while the icons get set trough a style property, like this:

```
<ListBoxItem>
    <ListBoxItem.Style>
        <Style TargetType="ListBoxItem" BasedOn="{StaticResource MaterialDesign3SmallNavigationRailPrimaryListBoxItem}">
            <Setter Property="materialDesign:ListBoxItemAssist.SelectedIcon" Value="Circle"/>
            <Setter Property="materialDesign:ListBoxItemAssist.UnselectedIcon" Value="CircleOutline"/>
        </Style>
    </ListBoxItem.Style>
    <TextBlock Text="Test 1"/>
</ListBoxItem>
```
Or like this with ItemSource Binding
```
<ListBox.ItemContainerStyle>
    <Style TargetType="ListBoxItem" BasedOn="{StaticResource MaterialDesign3NavigationRailPrimaryListBoxItem}">
        <Setter Property="materialDesign:ListBoxItemAssist.SelectedIcon" Value="{Binding SelectedIcon}"/>
        <Setter Property="materialDesign:ListBoxItemAssist.UnselectedIcon" Value="{Binding UnselectedIcon}"/>
    </Style>
</ListBox.ItemContainerStyle>
<ListBox.Resources>
    <DataTemplate DataType="{x:Type local:ItemSample}">
        <TextBlock Text="{Binding Title}"/>
    </DataTemplate>
</ListBox.Resources>
```

To set the NavigationRail in "Icon only mode" you can use the property:
`<Setter Property="materialDesign:ListBoxItemAssist.IsTextVisible" Value="False"/>`

The sample is under NavigationRail page in MaterialDesign3Demo app.

## Conclusion

So I'm waiting for feedback before cleaning up and in case making a Pull Request.
Also I'm not sure about the new Material Design 3 structure, for now the styles are in the standard MaterialDesignTheme.ListBox.xaml file. If it should go somewhere else, please let me know.

**Comment by MichelMichels** [comment]
This is nice work!

The style should go into MaterialDesign3.ListBox.xaml (you can see f.e. MaterialDesign3.TextBox,xaml already added).
As for the the icons in the ListBoxItem, it's a bit hairy but maybe we can have a better iteration in the future, I'm thinking maybe a custom control, but I don't know what @Keboo 's opinion is about this.
